src/pages/index.js

```diff
@@ -119,91 +117,154 @@ function getCardElement(data) {
   return cardElement;
 }
 
-function openModal(modal) {
-  modal.classList.add("modal_is-open");
-  document.addEventListener("keydown", closeOpenModalsOnEscape);
-}
+function handleLikeBtnClick(evt, id) {
+  evt.preventDefault();
 
-function closeModal(modal) {
-  modal.classList.remove("modal_is-open");
-  document.removeEventListener("keydown", closeOpenModalsOnEscape);
+  const likeBtnElement = evt.target;
+
+  api
+    .editLikeStatus({
+      cardId: id,
+      isLiked: !likeBtnElement.classList.contains("card__icon_active"),
+    })
+    .then(() => {
+      likeBtnElement.classList.toggle("card__icon_active");
+    })
+    .catch((err) => {
+      console.error("Error editing like status:", err);
+    });
 }
 
-modalElements.forEach((modal) => {
-  modal.addEventListener("click", (evt) => {
-    if (evt.target === modal) {
-      closeModal(modal);
-    }
-  });
-});
+function handleImageClick(data) {
+  previewModalImage.src = data.link;
+  previewModalImage.alt = data.name;
+  previewModalCaption.textContent = data.name;
 
-function closeOpenModalsOnEscape(evt) {
-  if (evt.key === "Escape") {
-    modalElements.forEach((modal) => {
-      if (modal.classList.contains("modal_is-open")) {
-        closeModal(modal);
-      }
-    });
-  }
+  openModal(previewModalElement);
 }
 
-editProfileBtn.addEventListener("click", function () {
-  profileNameInput.value = profileName.textContent;
-  profileDescriptionInput.value = profileDescription.textContent;
-  resetValidation(
-    editProfileModal,
-    [profileNameInput, profileDescriptionInput],
-    settings
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(deleteCardModal);
+}
+
+function handleDeleteCardFormSubmit(evt) {
+  handleSubmit(
+    () => {
+      return api.deleteCard(selectedCardId).then(() => {
+        selectedCard.remove();
+        closeModal(deleteCardModal);
+        return data;
```

Comment:
`        return data;
`
* When I delete a card, there is an error https://snipboard.io/4XBSn9.jpg 


It's not needed here